core/renderers/measurables/rows.js

```diff
@@ -541,3 +548,29 @@ Blockly.blockRendering.InputRow.prototype.measure = function() {
 Blockly.blockRendering.InputRow.prototype.endsWithElemSpacer = function() {
   return !this.hasExternalInput && !this.hasStatement;
 };
+
+
+/**
+ * pxt-blockly An object containing information about a row representing a
+ *   collapsed statement input.
+ * @param {!Blockly.blockRendering.ConstantProvider} constants The rendering
+ *   constants provider.
+ * @package
+ * @constructor
+ * @extends {Blockly.blockRendering.Row}
+ */
+Blockly.blockRendering.CollapsedInputRow = function(constants) {
```

Comment:
I would also move this into a pxt renderer measurable. See how its done in the zelos renderer: 
https://github.com/google/blockly/blob/master/core/renderers/zelos/measurables/row_elements.js